_search-plugins/search-pipelines/rerank-processor.md

```diff
@@ -110,9 +117,166 @@ POST /_search?search_pipeline=rerank_pipeline
 
 The `query_context` object contains the following fields. 
 
-Field name  | Description
-:--- | :---  
-`query_text` | The natural language text of the question that you want to use to rerank the search results. Either `query_text` or `query_text_path` (not both) is required.
-`query_text_path` | The full JSON path to the text of the question that you want to use to rerank the search results. Either `query_text` or `query_text_path` (not both) is required. The maximum number of characters in the path is `1000`.
+Field name | Required/Optional | Description
+:--- | :--- | :---  
+`query_text` | Required (see below)| The natural language text of the question that you want to use to rerank the search results. 
+`query_text_path` | Required (see below) | The full JSON path to the text of the question that you want to use to rerank the search results. The maximum number of characters in the path is `1000`.
+
+You are required to select either `query_text` or `query_text_path`, but not both simultaneously. Please ensure that only one option is chosen to avoid conflicts
+
+### The `by_field` rerank type
+
+The `by_field` rerank type is designed to operate using a specified field within your document. This is useful if a model has already ran and produced a numerical score in your document(s), or if a previous search response processor was applied and you want to rerank documents differently based on an aggregated field. For this rerank type, specify the following fields.
+
+Field  | Data type | Required/Optional | Description
+:--- | :---  | :--- | :--- 
+`by_field` | Object | Required | Provides the rerank processor with reranking information.
+`by_field.target_field` | String | Required |  This can be a path to your field that has a numerical score, in the form of `key[.key]`. 
```

Comment:
It's not clear if "that has" refers to "a path" or "your field". Please rephrase. No comma if "in the form of `key[.key]`" is essential information connected to "a numerical score". With the comma, the phrase can be applied to "a path".